crypto/src/main/java/org/springframework/security/crypto/password/DelegatingPasswordEncoder.java

```diff
@@ -286,9 +286,17 @@ public String encode(CharSequence rawPassword) {
 		@Override
 		public boolean matches(CharSequence rawPassword, String prefixEncodedPassword) {
 			String id = extractId(prefixEncodedPassword);
+			checkIfStringIsEmptyOrNull(id);
```

Comment:
I think it would improve readability to use `StringUtils.hasText`, use private constants, and then do:

```java
if (StringUtils.hasText(id)) {
    throw new IllegalArgumentException(String.format(NO_PASSWORD_ENCODER_MAPPED, id));
}
throw new IllegalArgumentException(NO_PASSWORD_ENCODER_PREFIX);
```

Reply:
Your idea was wonderful! I am pleased to inform you that your suggestion has been implemented in the source code.

Reply:
@jzheaux Sorry 

I couldn't rebase on 5.8.x. Should I clone branch 5.8.x, work on it, and create a new PR? Or is there another way?